### ISSUE TITLE
Odd bone lengths when blending between FK and IK

### ISSUE THREAD

**Issue by clamdragon** [kattkieru/grim_IK#1]
Hi Charles,
I recently started using the grim_vik node and it's pretty elegant. But I was getting odd bone lengths when blending between FK and IK - regardless of the stretch values in either mode. To condense to a thousand words, [this is what was happening](https://clamdragon3d.com/s/original.gif) (it can be pretty under-the-radar if the poses are very similar).

If that is the intended functionality then feel free to ignore this! But if [this](https://clamdragon3d.com/s/fixed.gif) is more like what it's supposed to do, then I've got a fork with the fix in place. Just let me know if you want to pull it.

**Comment by kattkieru** [comment]
Hi Brendan!

I'm surprised nobody else has mentioned that. :)

There was a thought initially and it had to do with stretching through blends (Especially with reverse or elbow pinning).  I haven't looked at your patch yet (I assume you're slerping the rotations), so I'm not sure how that works with all the separate features, but I'll have a look and let you know if it works here.  Are you cool with releasing your addition under the same license?

Thanks for using the node, btw, and for the feedback! If you use it on anything public please send a video. :)
~ Charles

**Comment by clamdragon** [comment]
Yes, same license is fine. I didn't actually include the updated binary I tested with (bit of a noob here when it comes to compiling c++, my .mll was about twice the size of the original so I'm sure there were some un-optimized settings).

The rotations were being slerp'd before I touched it, and I didn't change that. What I changed was the position interpolation - it was happening in world-space (in the "before" gif you can see the joints just draw a straight line from one pose to the next), so I just transformed to local-space and lerp'd _that_ before transforming back to world. Perhaps not the most efficient way to do that, but the dg profile was fine.
